lib/FreeRTOS/mqtt-agent/mqtt_agent.h

```diff
@@ -151,67 +157,72 @@ typedef void (* IncomingPublishCallback_t )( MQTTAgentContext_t * pMqttAgentCont
 
 /**
+ * @ingroup mqtt_agent_struct_types
  * @brief Information used by each MQTT agent. A context will be initialized by
  * MQTTAgent_Init(), and every API function will accept a pointer to the
  * initalized struct.
  */
 struct MQTTAgentContext
 {
-    MQTTContext_t mqttContext;
-    AgentMessageContext_t * pMessageCtx;
-    AckInfo_t pPendingAcks[ MQTT_AGENT_MAX_OUTSTANDING_ACKS ];
-    IncomingPublishCallback_t pIncomingCallback;
-    void * pIncomingCallbackContext;
-    bool packetReceivedInLoop;
+    MQTTContext_t mqttContext;                                 /**< MQTT connection information used by coreMQTT. */
+    AgentMessageContext_t * pMessageCtx;                       /**< Context used to deliver messages to the agent. */
+    AckInfo_t pPendingAcks[ MQTT_AGENT_MAX_OUTSTANDING_ACKS ]; /**< List of pending acknowledgment packets. */
+    IncomingPublishCallback_t pIncomingCallback;               /**< Callback to invoke for incoming publishes. */
+    void * pIncomingCallbackContext;                           /**< Context for incoming publish callback. */
+    bool packetReceivedInLoop;                                 /**< Whether a MQTT_ProcessLoop() call received a packet. */
 };
 
 /**
+ * @ingroup mqtt_agent_struct_types
  * @brief Struct holding arguments for a SUBSCRIBE or UNSUBSCRIBE call.
  */
 typedef struct MQTTAgentSubscribeArgs
 {
-    MQTTSubscribeInfo_t * pSubscribeInfo;
-    size_t numSubscriptions;
+    MQTTSubscribeInfo_t * pSubscribeInfo; /**< @brief List of MQTT subscriptions. */
+    size_t numSubscriptions;              /**< @brief Number of elements in `pSubscribeInfo`. */
 } MQTTAgentSubscribeArgs_t;
 
 /**
+ * @ingroup mqtt_agent_struct_types
  * @brief Struct holding arguments for a CONNECT call.
  */
 typedef struct MQTTAgentConnectArgs
 {
-    MQTTConnectInfo_t * pConnectInfo;
-    MQTTPublishInfo_t * pWillInfo;
-    uint32_t timeoutMs;
-    bool sessionPresent;
+    MQTTConnectInfo_t * pConnectInfo; /**< @brief MQTT CONNECT packet information. */
+    MQTTPublishInfo_t * pWillInfo;    /**< @brief Optional Last Will and Testament. */
+    uint32_t timeoutMs;               /**< @brief Maximum timeout for a CONNACK packet. */
+    bool sessionPresent;              /**< @brief Output flag set if a previous session was present. */
 } MQTTAgentConnectArgs_t;
 
 /**
+ * @ingroup mqtt_agent_struct_types
  * @brief Struct holding arguments that are common to every command.
  */
 typedef struct CommandInfo
 {
-    CommandCallback_t cmdCompleteCallback;
-    CommandContext_t * pCmdCompleteCallbackContext;
-    uint32_t blockTimeMs;
+    CommandCallback_t cmdCompleteCallback;          /**< @brief Callback to invoke upon completion. */
+    CommandContext_t * pCmdCompleteCallbackContext; /**< @brief Context for completion callback. */
+    uint32_t blockTimeMs;                           /**< @brief Maximum to block for enqueing the command. */
```

Comment:
```suggestion
    uint32_t blockTimeMs;                           /**< @brief Maximum time to block for enqueing the command. */
```

Reply:
Looks like both of us missed `enqueing`
```suggestion
    uint32_t blockTimeMs;                           /**< @brief Maximum time to block for enqueueing the command. */
```